spyder/preferences/shortcuts.py

```diff
@@ -214,118 +292,82 @@ def accept(self):
         self.button_ok.setFocus()
         super(ShortcutEditor, self).accept()
 
-    def keyPressEvent(self, e):
-        """Qt override."""
-        key = e.key()
-        # Check if valid keys
-        if key not in VALID_KEYS:
-            self.invalid_key_flag = True
-            return
-
-        self.npressed += 1
-        self.key_non_modifiers.append(key)
-        self.key_modifiers.add(key)
-        self.key_text.append(e.text())
-        self.invalid_key_flag = False
-
-        debug_print('key {0}, npressed: {1}'.format(key, self.npressed))
+    def event(self, e):
+        """Qt method override."""
+        if type(e) in [QEvent.Shortcut, QEvent.ShortcutOverride]:

```

Comment:
Technically, best practices would suggest making this a set (or at least a tuple), since its much faster and more conceptually linked to the use case, if my understanding is correct. Also, is there a reason for using ``type(e) in [...]`` instead of ``isinstance(e, (QEvent.Shortcut, QEvent.ShortcutOverride))``?

Reply:
I ported this from C++ Qt codebase, that's why. I'm going to make it more Pythonic.

Reply:
mmmh, your suggestion results in this error:
```TypeError: isinstance() arg 2 must be a type or tuple of types```
So I'm going to keep my formulation.